### ISSUE TITLE
2nd The Azure App Insights key is not a valid GUID (Code: InvalidBotData)

### ISSUE THREAD

**Issue by jocaffar** [OfficeDev/microsoft-teams-apps-company-communicator#975]
Same error as here: https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/964

GUIDs and app registrations have been regenerated unsuccessfully. 

---

{
  "code": "InvalidTemplateDeployment",
  "details": [
    {
      "details": [
        {
          "code": "InvalidBotData",
          "target": "XXXX-author",
          "message": "The Azure App Insights key is not a valid GUID"
        },
        {
          "code": "InvalidBotData",
          "target": "XXXX",
          "message": "The Azure App Insights key is not a valid GUID"
        }
      ]
    }
  ],
  "message": "The template deployment 'Microsoft.Template-20230228113616' is not valid according to the validation procedure. The tracking id is 'c8cf4142-0e11-4c36-9a8d-5c06571ac4cd'. See inner errors for details."
}

**Comment by vms87** [comment]
Hello. 
I have the same error. I have regenerated the GUIDs several times. Continue to fail

{
  "code": "InvalidTemplateDeployment",
  "details": [
    {
      "details": [
        {
          "code": "InvalidBotData",
          "target": "mycomunicator-author",
          "message": "The Azure App Insights key is not a valid GUID"
        },
        {
          "code": "InvalidBotData",
          "target": "mycomunicator",
          "message": "The Azure App Insights key is not a valid GUID"
        }
      ]
    }
  ],

**Comment by gsv022** [comment]
Hi @jocaffar ,
Thanks for raising the query. Please give us sometime and will get back to you with proper response soon.

**Comment by vms87** [comment]
Hello,
I have regenerated the GUIDs several times. Continue to fail...

**Comment by jocaffar** [comment]
Hi @gsv022. Yes, as @vms87 mentions, this was already tested several times unsuccessfully in several environments. As well as erasing and recreating app registrations.

**Comment by gsv022** [comment]
@jocaffar , We are checking on this issue and will get back to you with proper response soon.

**Comment by allengeorgek** [comment]
![image](https://user-images.githubusercontent.com/22046702/222067680-a4bd3df3-6265-465e-b152-9d327d4d78ba.png)

Tried to deploy the app for a customer today & received the following error similar to the ones being discussed. 
Generated new GUID & still the same error message while validating the Azure App Insights Key. Says it is not a valid GUID. 

Raw Error:
{  "code": "InvalidTemplateDeployment",  "details": [    {      "details": [        {          "code": "InvalidBotData",          "target": "TeamsCommunicator-author",          "message": "The Azure App Insights key is not a valid GUID"        },        {          "code": "InvalidBotData",          "target": "TeamsCommunicator",          "message": "The Azure App Insights key is not a valid GUID"        }      ]    }  ],  "message": "The template deployment 'Microsoft.Template-20230301122158' is not valid according to the validation procedure. The tracking id is '178caf13-3f5b-47d6-874a-c0d5eda0d90f'. See inner errors for details."}

**Comment by kitoes** [comment]
Hello, we have the same problem
We have already tried to regenerate the GUIDs but persists the problem

![image](https://user-images.githubusercontent.com/61427964/222177661-d95d2e67-5c3e-49ea-83c7-0d793f070a5b.png)

![image](https://user-images.githubusercontent.com/61427964/222177821-a711dccd-66ba-43bf-9590-a1f2d5730a50.png)

**Comment by v-royavinash** [comment]
@allengeorgek @kitoes We are prioritizing this issue as our highest concern, and we have tentatively identified the problem. Currently, we are testing a potential solution, and we will confirm once we have found a working fix.

**Comment by andrewmatveychuk** [comment]
> @allengeorgek @kitoes We are prioritizing this issue as our highest concern, and we have tentatively identified the problem. Currently, we are testing a potential solution, and we will confirm once we have found a working fix.

It might make sense to update the Application Insights resource definition in the deployment template as it uses not the up-to-date API version:
![image](https://user-images.githubusercontent.com/12231656/222223176-46aa1b27-2adf-4ccc-811a-17c9f90a0388.png)
Plus switching to the workspace-based AppInsights would be great considering the deprecation of the classic version.

**Comment by v-jaygupta** [comment]
@jocaffar @kitoes @andrewmatveychuk - We identified that the issue is not specific to the company communicator but is due to a recent update in either Bot Services or Azure template custom validation. The update is preventing the creation of a reference of App Insights Instrument key to the Microsoft Bot Service configuration. This change is impacting other templates and repositories that are dependent on it, including company communicator.
We have identified a possible workaround and are currently testing it with the highest priority. As soon as testing is complete, we will share the steps to unblock the deployment. Thank you for your patience and understanding as we work to resolve this issue.

**Comment by gsv022** [comment]
Hi Everyone,
- It is identified that the issue is not specific to the company communicator but is evident that there is an issue with Azure ARM resources functionality.
- We have updated the [Deploy to Azure](https://portal.azure.com/?feature.customportal=false/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FOfficeDev%2Fmicrosoft-teams-company-communicator-app%2Fmain%2FDeployment%2Fazuredeploy.json) link in 2nd step of the CC Deployment Guide. The earlier faced issue "_The Azure App Insights key is not a valid GUID_" is now resolved and deployment is working as expected.
- Please try deploying the Company communicator app and let us know if you face any further issues. 
- If the deployment is failing due to role assignments, please refer to the [Troubleshooting Guide](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting#4-multiple-instance-of-company-communicator-in-the-same-tenant-action-items).

**Comment by faranzazu** [comment]
> Hi Everyone,
> 
> * It is identified that the issue is not specific to the company communicator but is evident that there is an issue with Azure ARM resources functionality.
> * We have updated the [Deploy to Azure](https://portal.azure.com/?feature.customportal=false/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FOfficeDev%2Fmicrosoft-teams-company-communicator-app%2Fmain%2FDeployment%2Fazuredeploy.json) link in 2nd step of the CC Deployment Guide. The earlier faced issue "_The Azure App Insights key is not a valid GUID_" is now resolved and deployment is working as expected.
> * Please try deploying the Company communicator app and let us know if you face any further issues.
> * If the deployment is failing due to role assignments, please refer to the [Troubleshooting Guide](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting#4-multiple-instance-of-company-communicator-in-the-same-tenant-action-items).

The issue was fixed. Thank you!!!!!

**Comment by gsv022** [comment]
@jocaffar ,We are closing this request as we have provided the inputs for the query raised. If you face any issue with company communicator please raise a new issue for further assistance.

**Comment by jocaffar** [comment]
Hi @gsv022 thanks for your support. Feel free to close this issue :)
